Configuration for OTel support in iOS RUM (#16531)
DataDog/documentation: content/en/real_user_monitoring/connect_rum_and_traces/_index.md

-   Your HTTP servers are using [a library that supports distributed tracing](#supported-libraries).
-   You have the following set up based on your SDK:
    - With the **Browser SDK**, you have added the XMLHttpRequest (XHR) or Fetch resources on the RUM Explorer to your `allowedTracingUrls`.

>     - With the **Mobile SDK**, you have added the Native or XMLHttpRequest (XHR) to using `trackURLSession(firstPartyHosts:)` builder function.

Reviewer: ```suggestion
    - With the **Mobile SDK**, you have added the Native or XMLHttpRequest (XHR) to use `trackURLSession(firstPartyHosts:)` builder function.
```

      - `tracecontext`: [W3C Trace Context](https://www.w3.org/TR/trace-context/) (`traceparent`)
      - `b3`: [B3 single header](https://github.com/openzipkin/b3-propagation#single-header) (`b3`)
      - `b3multi`: [B3 multiple headers](https://github.com/openzipkin/b3-propagation#multiple-headers) (`X-B3-*`)
{{% /tab %}}

{{% tab "iOS RUM" %}}

> 1. Set up RUM to connect with APM as described above.

Reviewer: ```suggestion
1. [Set up RUM to connect with APM](#setup-rum).
```

Reviewer: Is that anchor link linking to the correct section?

Reviewer 2: If we change it here we should also change in the Browser section to make it coherent, no?

Reviewer: Oh yeah, okay, let's just leave it for now.

2. Use `trackURLSession(firstPartyHostsWithHeaderTypes:)` instead of `trackURLSession(firstPartyHosts:)` as follows:
    ```swift
    Datadog.initialize(
        appContext: .init(),
        configuration: Datadog.Configuration
            .builderUsing(
                rumApplicationID: "<rum_app_id>", 
                clientToken: "<client_token>", 
                environment: "<env_name>"
            )
            .trackURLSession(
                firstPartyHostsWithHeaderTypes: [
                    "api.example.com": [.tracecontext]
                ]
            )
            .build()
        )
    ```

>     `trackURLSession(firstPartyHostsWithHeaderTypes:)` takes `Dictionary<String, Set<TracingHeaderType>>` as a parameter, where key is a host and value is a list of supported tracing header types.

Reviewer: ```suggestion
    `trackURLSession(firstPartyHostsWithHeaderTypes:)` takes `Dictionary<String, Set<TracingHeaderType>>` as a parameter, where the key is a host and the value is a list of supported tracing header types.
```

>     `TracingHeaderType` in an enum representing following tracing header types:

Reviewer: ```suggestion
    `TracingHeaderType` in an enum representing the following tracing header types:
```
